api/azure/api.go

```diff
@@ -48,7 +51,7 @@ type ArtifactResource struct {
 type API interface {
 	HandleCheckRunEvent(*github.CheckRunEvent) (bool, error)
 	GetAzureArtifactsURL(owner, repo string, buildID int64) string
-	FetchAzureArtifact(BuildArtifact, io.Writer) error
+	FetchAzureArtifact(BuildArtifact, *multipart.Writer) error
```

Comment:
I would much prefer using `io.Writer` instead of any concrete writer type, because:

1. It is immediately obvious that this is a writer that doesn't need to close, as the standard `io` interfaces are well known, unlike the concrete writer types that can vary (e.g. a HTTP response writer needs to be closed).
2. It is easier to test by using mocks or other writers.

And it's also the convention that I've observed in the standard library.

Reply:
Reverted back to writer. It's not obvious to me whether creating multiple multipart writers into the same buffer will behave as expected.

Reply:
In fact, I didn't notice in your original version you called `CreateFormFile` inside the function, which means you did need a concrete type. I mistakenly thought you only needed `Write` and `Close`.